frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts

```diff
@@ -309,14 +309,57 @@ const getStackedAreasInterpolateTransform = (
   };
 };
 
-export const applySquareRootScaling = (value: RowValue): RowValue => {
-  if (typeof value === "number") {
-    const sign = value >= 0 ? 1 : -1;
-    return sign * Math.sqrt(Math.abs(value));
-  }
+const getSign = (value: number) => (value >= 0 ? 1 : -1);
+// TODO dedupe this function
+function signedSquareRoot(value: number) {
+  return getSign(value) * Math.sqrt(Math.abs(value));
+}
 
-  return value;
-};
+function getStackedPowerTransform(seriesDataKeys: DataKey[]): TransformFn {
```

Comment:
I think we should pass the axis transform here and use it for the stack calculation